src/test/unit/specific_feature_tests/dialogs.js

```diff
@@ -0,0 +1,465 @@
+        // close it
+        if (
+            dialogWindowOrFalse && (
+                typeof dialogWindowOrFalse.close === "function" ||
+                // IE
+                typeof dialogWindowOrFalse.close === "object"
+            )
+        ) {
+            dialogWindowOrFalse.close();
+        }
+    }
+
+    function assertOpened(dialogWindow) {
+        expect(expect() + 2);
+        if (sinonCantWrapWindowOpen !== true) {
+            expect(expect() + 1);
+            ok(
+                window.open.calledOnce,
+                "`window.open` was called"
+            );
+        }
+        ok(
+            dialogWindow.window === dialogWindow,
+            "Dialog window seems to exist"
+        );
+
+        strictEqual(
+            dialogWindow.document.title,
+            args.expectedTitle,
+            "Dialog title"
+        );
+    }
+    function assertNotOpened(returnedFalse) {
+        expect(expect() + 1);
+        if (sinonCantWrapWindowOpen !== true) {
+            expect(expect() + 1);
+            ok(
+                window.open.callCount === 0,
+                "`window.open` was not called"
+            );
+        }
+        ok(
+            returnedFalse === false,
+            "Dialog function returned `false`"
+        );
+    }
+}
+/* jshint latedef: true */
+
+dialogTestHelper.EXPECT_OPENED = "Expect this dialog to have been opened. " +
+    "Expect it with all of your heart.";
+dialogTestHelper.EXPECT_NOT = "Nope. The dialog should not have opened.";
+
+// Provided a dialog name, returns a jQuery selector for that dialog's button.
+function getDialogToolbarSelector(dialogName) {
+    if (typeof dialogName !== "string") {
+        throw "Expected a string";
+    }
+    return ".wym_section.wym_tools.wym_buttons li a[name=" + dialogName + "]";
+}
+
+function getDialogDocumentTitle(dialogName) {
+    if (typeof dialogName !== "string") {
+        throw "Expected a string";
+    }
+
+    var TITLES = {
+        CreateLink: "Link",
+        InsertImage: "Image",
+        Paste: "Paste from Word",
+        InsertTable: "Table",
+        Preview: "Preview"
+    };
+
+    if (TITLES.hasOwnProperty(dialogName) !== true) {
+        throw "No such dialog";
+    }
+    return TITLES[dialogName];
+}
+
+test("Link dialog doesn't open when no selection; by toolbar button click",
+    function () {
+        var dialogName = "CreateLink";
+        dialogTestHelper({
+            currentTest: this,
+            noChangeHtml: "<p>Foo</p>",
+            prepareFunc: function (wymeditor) {
+                wymeditor.deselect();
+            },
+            dialogButtonSelector: getDialogToolbarSelector(dialogName),
+            expectedOpenedOrNot: dialogTestHelper.EXPECT_NOT,
+            expectedTitle: getDialogDocumentTitle(dialogName)
+        });
+    }
+);
+
+test("Link dialog opens when selection is non-collapsed and " +
+    "`selectedContainer` is a non-anchor element; by toolbar button click",
```

Comment:
> I think it's important to point out that the selectedContainer isn't an anchor, here. So maybe:
and selectedContainer is a non-anchor element
instead of
and selectedContainer doesn't return false

@winhamwr here it is.